File: data/gcc_versions_comparison/src/program/src/func13.c
#include "program.h" 
array_t* func13(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array102;
   if (pCounter > 0) {
      array102 = vars->data[--pCounter];
      array102->refC++;
      DEBUG_COPY(array102->id);
   } else {
      array102 = (array_t*)malloc(sizeof(array_t));
      array102->size = 984;
      array102->refC = 1;
      array102->id = 102;
      array102->data = (unsigned int*)malloc(array102->size*sizeof(unsigned int));
      memset(array102->data, 0, array102->size*sizeof(unsigned int));
      DEBUG_NEW(array102->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array102;
   array_t* array103 = func17(&params0, loopsFactor);
   DEBUG_RETURN(array103->id);
   free(params0.data);
   if(PATH0 & 1) {
      array_t* array106;
      if (pCounter > 0) {
         array106 = vars->data[--pCounter];
         array106->refC++;
         DEBUG_COPY(array106->id);
      } else {
         array106 = (array_t*)malloc(sizeof(array_t));
         array106->size = 672;
         array106->refC = 1;
         array106->id = 106;
         array106->data = (unsigned int*)malloc(array106->size*sizeof(unsigned int));
         memset(array106->data, 0, array106->size*sizeof(unsigned int));
         DEBUG_NEW(array106->id);
      }
      unsigned int loop32 = 0;
      unsigned int loopLimit32 = (rand()%loopsFactor)/3 + 1;
      for(; loop32 < loopLimit32; loop32++) {
         for (int i = 0; i < array106->size; i++) {
            array106->data[i]--;
         }
         for (int i = 0; i < array103->size; i++) {
            if (array103->data[i] == 85) { 
               return array103;
            }
         }
      }
      array_t_param params1;
      params1.size = 3;
      params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
      params1.data[0] = array102;
      params1.data[1] = array103;
      params1.data[2] = array106;
      array_t* array107 = func28(&params1, rng(), loopsFactor);
      DEBUG_RETURN(array107->id);
      free(params1.data);
      array107->refC--;
      if(array107->refC == 0) {
         free(array107->data);
         free(array107);
         DEBUG_FREE(array107->id);
      }
      array106->refC--;
      if(array106->refC == 0) {
         free(array106->data);
         free(array106);
         DEBUG_FREE(array106->id);
      }
   }
   else {
      array_t_param params1;
      params1.size = 2;
      params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
      params1.data[0] = array102;
      params1.data[1] = array103;
      array_t* array108 = func22(&params1, rng(), loopsFactor);
      DEBUG_RETURN(array108->id);
      free(params1.data);
      array108->refC--;
      if(array108->refC == 0) {
         free(array108->data);
         free(array108);
         DEBUG_FREE(array108->id);
      }
   }
   array103->refC--;
   if(array103->refC == 0) {
      free(array103->data);
      free(array103);
      DEBUG_FREE(array103->id);
   }
   return array102;
}

File: data/multilang_comparison/program_nim/src/func13.nim
proc func13*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   var array198: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array198 = vars.data[pCounter]
      inc array198.refC
      debugCopy(array198.id)
   else:
      array198 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array198.size = 290
      array198.refC = 1
      array198.id = 198
      array198.data = cast[ptr UncheckedArray[uint32]](alloc(array198.size * sizeof(uint32)))
      zeroMem(array198.data, array198.size * sizeof(uint32))
      debugNew(array198.id)
   var params0: ArrayTParam
   params0.size = 1
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   params0.data[0] = array198
   var array199: ptr ArrayT = func17(addr params0, loopsFactor)
   debugReturn(array199.id)
   dealloc(params0.data)
   if (PATH0 and 1'u64) != 0:
      discard
      var array202: ptr ArrayT
      if pCounter > 0:
         dec pCounter
         array202 = vars.data[pCounter]
         inc array202.refC
         debugCopy(array202.id)
      else:
         array202 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
         array202.size = 886
         array202.refC = 1
         array202.id = 202
         array202.data = cast[ptr UncheckedArray[uint32]](alloc(array202.size * sizeof(uint32)))
         zeroMem(array202.data, array202.size * sizeof(uint32))
         debugNew(array202.id)
      let loopLimit59 = (50) div (3) + 1
      for loop59 in 0 ..< loopLimit59:
         discard
         for i in 0 ..< array198.size:
            dec array198.data[i]
         for i in 0 ..< array199.size:
            if array199.data[i] == 12:
               return array199
      var params1: ArrayTParam
      params1.size = 3
      params1.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params1.size * sizeof(ptr ArrayT)))
      params1.data[0] = array198
      params1.data[1] = array199
      params1.data[2] = array202
      var array203: ptr ArrayT = func28(addr params1, getPath(), loopsFactor)
      debugReturn(array203.id)
      dealloc(params1.data)
      dec array203.refC
      if array203.refC == 0:
         dealloc(array203.data)
         dealloc(array203)
         debugFree(array203.id)
      dec array202.refC
      if array202.refC == 0:
         dealloc(array202.data)
         dealloc(array202)
         debugFree(array202.id)
   else:
      discard
      var params1: ArrayTParam
      params1.size = 2
      params1.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params1.size * sizeof(ptr ArrayT)))
      params1.data[0] = array198
      params1.data[1] = array199
      var array204: ptr ArrayT = func22(addr params1, getPath(), loopsFactor)
      debugReturn(array204.id)
      dealloc(params1.data)
      dec array204.refC
      if array204.refC == 0:
         dealloc(array204.data)
         dealloc(array204)
         debugFree(array204.id)
   dec array198.refC
   if array198.refC == 0:
      dealloc(array198.data)
      dealloc(array198)
      debugFree(array198.id)
   return array199

